src/storage-types/src/connections.rs

```diff
@@ -443,6 +494,9 @@ impl KafkaConnection {
         storage_configuration: &StorageConfiguration,
         context: C,
         extra_options: &BTreeMap<&str, String>,
+        // Whether or not we are connecting from timely threads. If we are, IO will
+        // occur on Tokio tasks.
+        from_timely: bool,
```

Comment:
(nit, so feel free to ignore) Would it be more readable if this was like an enum like
```
enum AsyncRuntimeContext {
   Timely, Tokio
}
```

Reply:
ooh maybe!